hypothesis-python/tests/common/debug.py

```diff
@@ -97,7 +97,7 @@ def assert_all_examples(strategy, predicate, settings=None):
     """
 
     @given(strategy)
-    @Settings(parent=settings)
+    @Settings(parent=settings, database=None)
```

Comment:
I suspect this may be the cause of several `data_too_large` flakes I've seen recently.

With #3862, this would have stopped flaking, but still do needless replays.

If we invalidate databases, we've already discussed adding `__qualname__` for a slight improvement in discriminating tests. Maybe we should add the strategy `repr` as well, to cover this kind of wrapped tests?

Reply:
Hmm, yeah, I can see how that'd help.  We could make it more precise by applying that logic only to tests which are indented but _don't_ accept a `self` argument.

And we actually invalidate database keys pretty regularly for specific strategies, and I'm comfortable with a global-invalidation if our users get something out of it too.  It's nice if we can bundle those into a few releases per year ofc, but I'd only suggest delaying such a release if we were actively working on another such change.

Reply:
I don't have time to follow up on this now I'm afraid. I'll write up an issue to not forget it. Then, if no-one beats me to it, I might pick it up at a later date with a PR (to be merged at an opportune time). Sounds ok?